src/gluonts/nursery/streaming_models/predictor.py

```diff
@@ -0,0 +1,145 @@
+        with (path / "type.txt").open("w") as fp:
+            fp.write(fqname_for(self.__class__))
+
+    @classmethod
+    def deserialize(cls, path: Path, *args, **kwargs) -> "StreamPredictor":
+        with (path / "type.txt").open("r") as fp:
+            tpe_str = fp.readline()
+            tpe = locate(tpe_str)
+
+        assert isinstance(tpe, type)
+        assert issubclass(tpe, StreamPredictor)
+
+        return tpe.deserialize(path, *args, **kwargs)  # type: ignore
+
+
+class StateAwarePredictorWrapper(StreamPredictor):
+    """
+    A wrapper around GluonTS predictors that are "state aware", i.e. that
+    consume a state field when processing a data entry, and include the
+    update state in the returned forecasts (in the "info" field).
+
+    Parameters
+    ----------
+    predictor
+        The GluonTS predictor object to be wrapped.
+    state_initializer
+        A function with no argument, that returns the initial state for the predictor.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        predictor: Predictor,
+        state_initializer: Callable[[], Any],
+    ) -> None:
+        # TODO: this should rather be undefined (None? +infty?) because there's
+        # TODO: no concept of "prediction" length in the streaming case
+        assert predictor.prediction_length == 1
+
+        self.predictor = predictor
+        self.state_initializer = state_initializer
+
+    @property
+    def lead_time(self) -> int:
+        return self.predictor.lead_time
+
+    def initial_state(self) -> StreamState:
+        return self.state_initializer()
+
+    def step(
+        self, data: DataEntry, state: StreamState
+    ) -> Tuple[DistributionForecast, StreamState]:
+        data_and_state = {**data, **state}
+
+        # TODO: the following two lines could be nicer, if only predictors
+        # TODO: allowed for a 1-item prediction directly
+        forecasts = self.predictor.predict([data_and_state])
+        forecast = next(iter(forecasts))
+
+        assert isinstance(forecast, DistributionForecast)
+
+        return forecast, forecast.info[PREDICTOR_STATE_KEY]
+
+    def serialize(self, path: Path) -> None:
+        (path / "predictor").mkdir(parents=True, exist_ok=True)
+        self.predictor.serialize(path / "predictor")
+
+        super().serialize(path)
+        with (path / "wrapper_parameters.json").open("w") as fp:
+            parameters = dict(
+                lead_time=self.lead_time,
+                state_initializer=self.state_initializer,
+            )
+            print(dump_json(parameters), file=fp)
+
+    @classmethod
+    def deserialize(
+        cls,
+        path: Path,
+        ctx: Optional[mx.Context] = None,
+        *args,
+        **kwargs,
+    ) -> "StateAwarePredictorWrapper":
+        ctx = ctx if ctx is not None else get_mxnet_context()
+        with mx.Context(ctx):
+            predictor = Predictor.deserialize(path / "predictor", ctx)  # type: ignore
```

Comment:
This should fail. The `ctx` should be removed (this was for the 0.4.x gluonts). I guess serde is not captured in the included tests that's why there was no issue.

Reply:
Just saw this, I'll fix it